### ISSUE TITLE
MathML printer for MatrixElement

### ISSUE THREAD

**Issue by oscargus** [sympy/sympy#15974]
Currently

```
from sympy import mathml, MatrixSymbol
A = MatrixSymbol('A', 3, 3)
print(mathml(A[0,0]))
print(mathml(A[0,0], printer='presentation'))
```

gives an error as it falls back to `Basic`:
```
Traceback (most recent call last):

  File "<ipython-input-1-05d31a862058>", line 3, in <module>
    print(mathml(A[0,0]))

  File "C:\Users\Oscar\sympy\sympy\printing\mathml.py", line 916, in mathml
    return MathMLContentPrinter(settings).doprint(expr)

  File "C:\Users\Oscar\sympy\sympy\printing\mathml.py", line 61, in doprint
    mathML = Printer._print(self, expr)

  File "C:\Users\Oscar\sympy\sympy\printing\printer.py", line 287, in _print
    return getattr(self, printmethod)(expr, **kwargs)

  File "C:\Users\Oscar\sympy\sympy\printing\mathml.py", line 440, in _print_Basic
    x.appendChild(self._print(arg))

  File "C:\Users\Oscar\sympy\sympy\printing\printer.py", line 287, in _print
    return getattr(self, printmethod)(expr, **kwargs)

  File "C:\Users\Oscar\sympy\sympy\printing\mathml.py", line 440, in _print_Basic
    x.appendChild(self._print(arg))

  File "C:\Users\Oscar\Anaconda3\lib\xml\dom\minidom.py", line 114, in appendChild
    if node.nodeType == self.DOCUMENT_FRAGMENT_NODE:

AttributeError: 'str' object has no attribute 'nodeType'
```

After #15891 is merged it at least produces an output, but not correctly:
```
<matrixelement><ci>A</ci><cn>0</cn><cn>0</cn></matrixelement>
<mrow><mi>matrixelement</mi><mfenced><mi>A</mi><mn>0</mn><mn>0</mn></mfenced></mrow>
```

Hence, there are two problems to be fixed:
 - [x] Correct `_print_Basic` where the string should be embedded in `self.dom.createTextNode` or something like that
 - [x] Add a dedicated printer, `_print_MatrixElement`, leading to a slightly better result than this. For example, the LaTeX printer uses subscripts for the indices. Parts of `_print_Symbol` can be used as inspiration or maybe even called cleverly.

**Comment by sashank27** [comment]
Hi! I would like to work on this issue. Let me know how to proceed on this.

**Comment by oscargus** [comment]
For the first problem you will just have to debug the execution and see what the conditions are for it to go wrong. The `_print_Basic` method works for many cases, but apparently not for this. I would expect it to recursively call itself and the second time it does not work based on the traceback. Hence, you will need to figure out what the input argument is to this second call and probably generate code in some other way. I guess that the reason that it works after #15891 is that the first call of the MatrixSymbol now goes to a dedicated method, not the Basic-printer again.

For the second problem:

Look at the `_print_Symbol` code for MathML and try to figure out how it works for subscripts. Basically you will need to implement a very similar code, but you know that there will be exactly two subscripts here.

To see what is expected as an output, you can just create a `Symbol` with the name `A_1_1` or something similar. This should result in about the correct MathML (except that the current code is not 100% correct, see #15718 ...).

**Comment by oscargus** [comment]
There is a fix for correct sub- and superscripts in #15976

**Comment by Guvvala-Prasanth-Reddy** [comment]
Hi @sashank27 !  are you still  working on this if not I will take it?

**Comment by oscargus** [comment]
For the information of both @sashank27 and @Guvvala-Prasanth-Reddy : Both the PRs mentioned above are now merged, so not clear how to solve the `Basic` printing problem, if it still exists. I guess one will need to find a new class that also falls back on Basic and provides an error, so maybe this is not the highest priority.

For the actual MatrixElement printer have a look at the `_print_Symbol` code to see how to properly generate subscripts.

**Comment by shubhamabhang77** [comment]
Is there anything working on this right now? I would like to work on this issue.

**Comment by oscargus** [comment]
@shubhamabhang77 I do not know the status of @sashank27 and @Guvvala-Prasanth-Reddy .

However, if you want to work on more MathML stuff, I can recommend #6013 but for MathML (should be remapping lerchphi to the correct Phi-sign, polylog prints as zeta now, but `polylog(s, z)` should print as Li_s(z) (where _s is sub-s), see https://en.wikipedia.org/wiki/Polylogarithm ).
![image](https://user-images.githubusercontent.com/8114497/53200285-7a964580-3621-11e9-95a9-59045ee138b0.png)


I'm also about to create a new issue with MathML printing functions that are currently not supported.

**Comment by arooshiverma** [comment]
Hey ! Is anyone working on this? I want to do so if no one is. Please let me know.

**Comment by oscargus** [comment]
Not that I am aware of, so please go ahead. (More MathML functions in #16036 I put you up on this one for now.)

**Comment by oscargus** [comment]
Here is an example of what it should look like (the LaTeX column) and what it currently looks like (the MathML column). The gray area is the expression, but see more below.
![image](https://user-images.githubusercontent.com/8114497/53943117-b1a63580-40bc-11e9-8115-e91b688e4899.png)

The first one is
```
X = MatrixSymbol('X', 10, 10)
X[0, 0]
```
so the `MatrixElement` at 0, 0 from a 10 by 10 matrix.

I also included an example of `MatrixSlice` which is related in case you want to do that as well.
```
X[:5, 1:9:2]
``` 
is the call leading to this `MatrixSlice`.

**Comment by oscargus** [comment]
A good idea is to look at the corresponding code in the LaTeX printer and try to follow the same logic.

**Comment by arooshiverma** [comment]
I am getting the following error

>>> from sympy import mathml,MatrixSymbol
>>> A=MatrixSymbol('A', 3, 3)
>>> print(mathml(A[0,0]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\aroos\Anaconda2\lib\site-packages\sympy\printing\mathml.py", line 905, in mathml
    return MathMLContentPrinter(settings).doprint(expr)
  File "C:\Users\aroos\Anaconda2\lib\site-packages\sympy\printing\mathml.py", line 50, in doprint
    mathML = Printer._print(self, expr)
  File "C:\Users\aroos\Anaconda2\lib\site-packages\sympy\printing\printer.py", line 287, in _print
    return getattr(self, printmethod)(expr, **kwargs)
  File "C:\Users\aroos\Anaconda2\lib\site-packages\sympy\printing\mathml.py", line 429, in _print_Basic
    for arg in e:
TypeError: 'MatrixElement' object is not iterable

**Comment by oscargus** [comment]
That is partly why a special printer is needed. However, there are two different MathML printers. The presentation one is used as 
`print(mathml(A[0,0], printer='presentation'))`
